File: php/features.php
<?php
declare(strict_types=1);

// EnergyCharts SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class EnergyChartsFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new EnergyChartsBaseFeature();
            case "test":
                return new EnergyChartsTestFeature();
            default:
                return new EnergyChartsBaseFeature();
        }
    }
}

File: py/features.py
# EnergyCharts SDK feature factory

from feature.base_feature import EnergyChartsBaseFeature
from feature.test_feature import EnergyChartsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: EnergyChartsBaseFeature(),
        "test": lambda: EnergyChartsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
